src/software/utils/main_mergeMeshes.cpp

```diff
@@ -0,0 +1,309 @@
+           "* boolean_intersection: Create a new mesh from the intersected volumes of the two input meshes.\n"
+           "* boolean_difference: Create a new mesh from the volume of the first input mesh substracted by the second input mesh.\n";
+}
+
+/**
+ * @brief convert an enum EOperationType to its corresponding string
+ * @param EOperationType
+ * @return String
+ */
+std::string EOperationType_enumToString(EOperationType operationType)
+{
+  switch(operationType)
+  {
+      case EOperationType::BOOLEAN_UNION:         return "boolean_union";
+      case EOperationType::BOOLEAN_INTERSECTION:  return "boolean_intersection";
+      case EOperationType::BOOLEAN_DIFFERENCE:    return "boolean_difference";
+  }
+  throw std::out_of_range("Invalid operationType enum: " + std::to_string(int(operationType)));
+}
+
+/**
+ * @brief convert a string operationType to its corresponding enum EOperationType
+ * @param String
+ * @return EOperationType
+ */
+EOperationType EOperationType_stringToEnum(const std::string& operationType)
+{
+  std::string type = operationType;
+  std::transform(type.begin(), type.end(), type.begin(), ::tolower); // tolower
+
+  if(type == "boolean_union")         return EOperationType::BOOLEAN_UNION;
+  if(type == "boolean_intersection")  return EOperationType::BOOLEAN_INTERSECTION;
+  if(type == "boolean_difference")    return EOperationType::BOOLEAN_DIFFERENCE;
+  throw std::out_of_range("Invalid operationType: " + operationType);
+}
+
+std::ostream& operator<<(std::ostream& os, const EOperationType operationType)
+{
+    os << EOperationType_enumToString(operationType);
+    return os;
+}
+
+std::istream& operator>>(std::istream& in, EOperationType& operationType)
+{
+    std::string token;
+    in >> token;
+    operationType = EOperationType_stringToEnum(token);
+    return in;
+}
+
+/**
+ * @brief Pre/Post-processes a mesh for boolean operations.
+ * @details Triangulates the facets, collapses the small edges
+ *  and removes self-intersections.
+ *
+ * @ from Geogram library boolean operation examples
+ */
+void fixMeshForBooleanOperations(GEO::Mesh& m)
+{
+    GEO::mesh_repair(m, GEO::MeshRepairMode(GEO::MESH_REPAIR_COLOCATE | GEO::MESH_REPAIR_DUP_F), 1e-3 * GEO::surface_average_edge_length(m));
+    GEO::tessellate_facets(m, 3);
+    GEO::mesh_remove_intersections(m);
+}
+
+/**
+ * @brief Merge two meshes
+ */
+int aliceVision_main(int argc, char** argv)
+{
+    // timer initialization
+
+    system::Timer timer;
+
+    // command-line parameters
+
+    std::string verboseLevel = system::EVerboseLevel_enumToString(system::Logger::getDefaultVerboseLevel());
+    std::string inputFirstMeshPath;
+    std::string inputSecondMeshPath;
+    std::string outputFilePath;
+    std::string operationTypeName = EOperationType_enumToString(EOperationType::BOOLEAN_UNION);
+
+    bool preProcess = true;
+    bool postProcess = true;
+
+    po::options_description allParams("AliceVision mergeMeshes");
```

Comment:
Please specify better what the program does e.g. "The program takes two meshes and..."

Reply:
done, thanks